### PR TITLE
Fix: make fill() and fill_n() work with lvalues

### PR DESCRIPTION
**Pull request by Kijewski**
repeat_n was a combination of repeat and take with no advantages over keeping them separate. Instead of fixing the problem is two places, I removed the simply removed repeat_n.

### CODE REVIEW

File: include/rx/ranges.hpp [hunk 1]
```diff
@@ -735,7 +735,8 @@ seq(T, U)->seq<T, U>;
 template <class T>
 struct repeat {
     T value;
-    constexpr explicit repeat(T value) noexcept : value(std::move(value)) {}
+    template <class V>
+    constexpr explicit repeat(V&& value) noexcept : value(std::forward<V>(value)) {}
```
Comment (Hazurl) [inline include/rx/ranges.hpp]: Prevent use of copy/move constructor

**Comment by simonask** [comment]
I think it makes sense to express `repeat_n` as a combination of `repeat` and `take`, but I would not remove it entirely. Just make it a function retuning `repeat | take`. :-)

**Comment by Kijewski** [comment]
repeat_n was just another name for fill_n. fill_n still exists. :) Should I re-add repeat_n as an alias?

**Comment by simonask** [comment]
Let's settle on either `repeat` or `fill`, not both — `fill` may be less surprising given `std::fill`, so I'd be fine with that. :-)

So then only `fill` and `fill_n`, no `repeat`/`repeat_n`.

**Comment by Kijewski** [comment]
Please have a look.

**Comment by simonask** [comment]
Looks good to me 👌  I'll merge right away, and update the README accordingly. :-)

**Pull request merged**